src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.java

```diff
@@ -56,50 +57,62 @@
+
+    @Inject private DialogService dialogService;
+    @Inject private PreferencesService preferencesService;
+    @Inject private TaskExecutor taskExecutor;
+
+    private LinkedFilesEditorViewModel viewModel;
 
     private ObservableOptionalValue<BibEntry> bibEntry = EasyBind.wrapNullable(new SimpleObjectProperty<>());
+    private final UiThreadObservableList<LinkedFileViewModel> decoratedModelList;
 
     public LinkedFilesEditor(Field field,
-                             DialogService dialogService,
                              BibDatabaseContext databaseContext,
-                             TaskExecutor taskExecutor,
                              SuggestionProvider<?> suggestionProvider,
-                             FieldCheckers fieldCheckers,
-                             PreferencesService preferences) {
-        this.viewModel = new LinkedFilesEditorViewModel(field, suggestionProvider, dialogService, databaseContext, taskExecutor, fieldCheckers, preferences);
-        this.dialogService = dialogService;
+                             FieldCheckers fieldCheckers) {
+        this.field = field;
         this.databaseContext = databaseContext;
-        this.preferencesService = preferences;
+        this.suggestionProvider = suggestionProvider;
+        this.fieldCheckers = fieldCheckers;
 
         ViewLoader.view(this)
                   .root(this)
                   .load();
 
-        ViewModelListCellFactory<LinkedFileViewModel> cellFactory = new ViewModelListCellFactory<LinkedFileViewModel>()
+        decoratedModelList = new UiThreadObservableList<>(viewModel.filesProperty());
+        Bindings.bindContentBidirectional(listView.itemsProperty().get(), decoratedModelList);
+    }
+
+    @FXML
+    private void initialize() {
+        this.viewModel = new LinkedFilesEditorViewModel(field, suggestionProvider, dialogService, databaseContext, taskExecutor, fieldCheckers, preferencesService);
+
+        new ViewModelListCellFactory<LinkedFileViewModel>()
                 .withStringTooltip(LinkedFileViewModel::getDescriptionAndLink)
                 .withGraphic(this::createFileDisplay)
                 .withContextMenu(this::createContextMenuForFile)
                 .withOnMouseClickedEvent(this::handleItemMouseClick)
                 .setOnDragDetected(this::handleOnDragDetected)
                 .setOnDragDropped(this::handleOnDragDropped)
                 .setOnDragOver(this::handleOnDragOver)
-                .withValidation(LinkedFileViewModel::fileExistsValidationStatus);
-
-        listView.setCellFactory(cellFactory);
+                .withValidation(LinkedFileViewModel::fileExistsValidationStatus)
+                .install(listView);
         listView.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
 
-        decoratedModelList = new UiThreadObservableList<>(viewModel.filesProperty());
-        Bindings.bindContentBidirectional(listView.itemsProperty().get(), decoratedModelList);
```

Comment:
Why was this removed?

Reply:
Its not, it stayed in the constructor, while other initialization stuff is moved to the initialize method.

Reply:
ah I see!  Github diff was a bit weird at that point